android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -615,6 +625,46 @@ public void disableNativeFramesTracking() {
         }
     }
 
+    public void startProfiling(Promise promise) {
+        final WritableMap result = new WritableNativeMap();
+        try {
+            RNSentryModuleImpl.javaScriptExecutorFactory.startSamplingProfiler();
+        } catch (UnsupportedOperationException e) {
+            String error = javaScriptExecutorFactory.toString() + "does not support Sampling Profiler";
+            logger.log(SentryLevel.ERROR, error);
+            result.putString("error", error);
+        }
+        promise.resolve(result);
+    }
+
+    public void stopProfiling(Promise promise) {
+        final WritableMap result = new WritableNativeMap();
+        try {
+            final File output = File.createTempFile(
+        "sampling-profiler-trace", ".cpuprofile", reactApplicationContext.getCacheDir());
+            RNSentryModuleImpl.javaScriptExecutorFactory.stopSamplingProfiler(output.getPath());
+
+            StringBuilder text = new StringBuilder();
```

Comment:
L: Would make sense to use `StringBuffer` as well?

Reply:
Do you mean for the file name?

Reply:
Nope, I meant instead `StringBuilder` sine you are using Buffer everywhere.

Reply:
If you dont need thread safety, `StringBuilder` is fine and faster.

Reply:
I'll keep `StringBuilder` as we don't need thread safety in this case.